src/hotspot/cpu/x86/x86.ad

```diff
@@ -8615,56 +8629,85 @@ instruct vmuladdaddS2I_reg(vec dst, vec src1, vec src2) %{
 
 // --------------------------------- PopCount --------------------------------------
 
-instruct vpopcountI_popcntd(vec dst, vec src) %{
-  predicate(VM_Version::supports_avx512_vpopcntdq());
+instruct vpopcount_integral_reg_evex(vec dst, vec src) %{
+  predicate((((Matcher::vector_element_basic_type(n->in(1)) == T_INT ||
+               Matcher::vector_element_basic_type(n->in(1)) == T_LONG)) &&
+             VM_Version::supports_avx512_vpopcntdq()) ||
+            (is_subword_type(Matcher::vector_element_basic_type(n->in(1))) &&
+             VM_Version::supports_avx512_bitalg()));
```

Comment:
This sort of check is happening multiple times. Should we have a function which takes bt and tells us if vpopcnt instruction is supported for the type.

Reply:
This is only in couple of predicates, so it think it will not hurt much.

Reply:
It is about maintenance and review.